hpx/runtime/threads/thread_data.hpp

```diff
@@ -243,7 +243,7 @@ namespace hpx { namespace threads {
 
     public:
         /// Return the id of the component this thread is running in
-#if defined(HPX_GCC_VERSION) && (HPX_GCC_VERSION >= 70000)
+#if defined(HPX_GCC_VERSION) && (HPX_GCC_VERSION >= 70300)
```

Comment:
This disables constexpr for anything but gcc > V7.3, is that intended?

Reply:
Very good catch... not at all intended. @khuck I can push something on Monday if you'd like, unless you get to it first.